src/prefect/_internal/concurrency/api.py

```diff
@@ -231,6 +234,25 @@ def wait_for_call_in_loop_thread(
         done_callbacks: Optional[Iterable[Call]] = None,
         contexts: Optional[Iterable[ContextManager]] = None,
     ) -> Awaitable[T]:
+        # Check for a running event loop to prevent blocking
+        if get_running_loop():
+            if greenback.has_portal():
+                # Use greenback to avoid blocking the event loop while waiting
+                return greenback.await_(
+                    from_async.wait_for_call_in_loop_thread(
+                        __call,
+                        timeout=timeout,
+                        done_callbacks=done_callbacks,
+                        contexts=contexts,
+                    )
+                )
+            else:
+                logger.error(
+                    "Detected unsafe call to `from_sync` from thread with event loop. "
+                    "Use `await greenback.ensure_portal()` to allow call to run "
+                    "without blocking the event loop."
+                )
```

Comment:
I think this should never happen in production, just during internal changes. Maybe we should throw an exception here?